src/websocket/WebsocketServer.ts

```diff
@@ -191,19 +194,8 @@ export class WebsocketServer extends EventEmitter {
                         }
 
                         try {
-                            const handler = this.requestHandlers.getInstance(request.type)
-                            if (handler) {
-                                logger.debug('socket "%s" sent request "%s" with contents "%o"', connection.id, request.type, request)
-                                handler.call(this.requestHandlers, connection, request)
-                            } else {
-                                connection.send(new ControlLayer.ErrorResponse({
-                                    version: request.version,
-                                    requestId: request.requestId,
-                                    errorMessage: `Unknown request type: ${request.type}`,
-                                    // @ts-expect-error
-                                    errorCode: 'INVALID_REQUEST',
-                                }))
-                            }
+                            logger.debug('socket "%s" sent request "%s" with contents "%o"', connection.id, request.type, request)
+                            await this.requestHandler.handleRequest(connection, request)
                         } catch (err) {
                             connection.send(new ControlLayer.ErrorResponse({
```

Comment:
Because an `await`  has been added above, I believe we should add a if-guard before calling `connection.send`:

```ts
if (connection.isDead()) {
    return
}
```

Reply:
Fixed